omegaconf/base.py

```diff
@@ -336,6 +313,51 @@ def _select_impl(
         )
         return root, last_key, value
 
+    def _resolve_complex_interpolation(
+        self,
+        parent: Optional["Container"],
+        value: "Node",
+        key: Any,
+        parse_tree: OmegaConfGrammarParser.ConfigValueContext,
+        throw_on_missing: bool,
+        throw_on_resolution_failure: bool,
+    ) -> Optional["Node"]:
+        """
+        A "complex" interpolation is any interpolation that cannot be handled by
+        `resolve_simple_interpolation()`, i.e. that either contains nested
+        interpolations or is not a single "${..}" block.
+        """
+
+        from .nodes import StringNode
+
+        value_str = value._value()
+        assert isinstance(value_str, str)
+
+        visitor = GrammarVisitor(
+            container=self,
+            resolve_args=dict(
```

Comment:
1. What the extra dict here instead of passing the arguments directly?
2. Is this object expensive to initialize?

Reply:
1. This code has changed already, see https://github.com/omry/omegaconf/pull/321#discussion_r494602651

2. Instantiating a `GrammarVisitor` is very cheap (just a few attribute assignments, and the `super()` constructor currently does nothing)